### PR TITLE
docs: Java extension settings

### LINKED ISSUE #18419: doc: add the info about `zed-java-eclipse-jdtls`
### Check for existing issues

- [X] Completed

### Describe the feature

There is no LSP supports reported on https://zed.dev/docs/languages/java, and [`zed-java-eclipse-jdtls`](https://github.com/ABckh/zed-java-eclipse-jdtls) claims it has added support for java language servers (eclipse-jdtls).

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by HeavySnowJakarta**
Closes #18419

Release Notes:

- updated `java`'s lsp support
- added `zed-java-eclipse-jdtls`
- copied the both's docs for configuration

@valentinegb @louisnicolas-longheval-vinci @maxdeviant @ABckh

### CODE REVIEW

**Comment by HeavySnowJakarta** [comment]
i have no `pnpm` installed. gonna try github codespace

**Comment by HeavySnowJakarta** [comment]
now it should passes the format check @notpeter

**Comment by valentinegb** [comment]
You might want to hold off for, maybe a day, since zed-extensions/java has a branch, specifically `initialization-options` which changes the configuration section of the `README.md` a moderate amount. I plan to merge it very soon, but for now I'm going to sleep ^^'

File: docs/src/languages/java.md [hunk 1]
```diff
@@ -1,6 +1,80 @@
...
-Java language support in Zed is provided the [zed Java extension](https://github.com/zed-extensions/java).
-Report issues to: [https://github.com/zed-extensions/java/issues](https://github.com/zed-extensions/java/issues)
+Java language support in Zed is provided the [zed Java extension](https://github.com/zed-extensions/java) and [zed-java-eclipse-jdtls](https://github.com/ABckh/zed-java-eclipse-jdtls). You may choose one between them to satisfy your needs.
+
+## Java extension for Zed
+
+- Tree Sitter: [tree-sitter/tree-sitter-java](https://github.com/tree-sitter/tree-sitter-java)
+- Language Server: [eclipse-jdtls/eclipse.jdt.ls](https://github.com/eclipse-jdtls/eclipse.jdt.ls)
+
+Report issues to: [https://github.com/zed-extensions/java/issues](https://github.com/zed-extensions/java/issues).
```
Comment (valentinegb) [inline docs/src/languages/java.md:10]: Can't this be...

```suggestion
Report issues to: <https://github.com/zed-extensions/java/issues>.
```

...instead?

**Comment by valentinegb** [comment]
> You might want to hold off for, maybe a day, since [zed-extensions/java](https://github.com/zed-extensions/java) has a branch, specifically `initialization-options` which changes the configuration section of the `README.md` a moderate amount. I plan to merge it very soon, but for now I'm going to sleep ^^'

I don't think I can merge yet. The **Initialization Options** section of the `README.md` in that branch is accurate even for the currently released version, though, so I would recommend at least adding that now.

**Comment by valentinegb** [comment]
[v2.0.0](https://github.com/zed-industries/extensions/pull/1519) has now been released and with it I expect the `README.md` to stay how it is for a good while.

**Comment by HeavySnowJakarta** [comment]
done. it should passes the format check now @notpeter

**Comment by HeavySnowJakarta** [comment]
a little weird as i just run the command of `pnpm prettier` after the commit. maybe there's sth wrong with what i did?

File: docs/src/languages/java.md [hunk 2]
```diff
@@ -1,6 +1,170 @@
 # Java
 
-Java language support in Zed is provided the [zed Java extension](https://github.com/zed-extensions/java).
-Report issues to: [https://github.com/zed-extensions/java/issues](https://github.com/zed-extensions/java/issues)
+Java language support in Zed is provided the [zed Java extension](https://github.com/zed-extensions/java) and [zed-java-eclipse-jdtls](https://github.com/ABckh/zed-java-eclipse-jdtls). You may choose one between them to satisfy your needs.
+
+## Java extension for Zed
 
 - Tree Sitter: [tree-sitter/tree-sitter-java](https://github.com/tree-sitter/tree-sitter-java)
+- Language Server: [eclipse-jdtls/eclipse.jdt.ls](https://github.com/eclipse-jdtls/eclipse.jdt.ls)
```
Comment (valentinegb) [inline docs/src/languages/java.md:8]: Sorry, but I just noticed that this line is different from line 126 even though both extensions use the same language server.

**Comment by notpeter** [comment]
I took a pass at cleaning this up and adding some instructions for install openjdk and jdtls.

Feel free to make to submit follow-on PRs, I'm sure I got some stuff wrong.
I just wanted to get something published more than this ASAP:
Screenshot 2024-10-17 at 0 06 39

https://zed.dev/docs/languages/java

Thanks all!

**Comment by valentinegb** [comment]
Thank you lots @notpeter! :heart:

There are just a few little things I think could be adjusted (again, I really do appreciate you getting this merged @notpeter, I agree with your reasoning):

- While you *can* use a version of JDTLS you install yourself for the Java with Eclipse JDTLS extension, that extension (only) *does* install a version of JDTLS for users.
- A completely understandable error, but the [JDTLS Language Server Settings & Capabilities](https://github.com/eclipse-jdtls/eclipse.jdt.ls/wiki/Language-Server-Settings-&-Capabilities) wiki page is *not* actually for initialization options (and the `settings` field is defined by the extension, not the language server). The correct reference would be [this section of the "Running the JAVA LS server from the command line" page](https://github.com/eclipse-jdtls/eclipse.jdt.ls/wiki/Running-the-JAVA-LS-server-from-the-command-line#initialize-request).
- The header on line 104 is *probably* a higher level than it should be, and the header on line 187 is probably a *lower* level than it should be.

**Comment by valentinegb** [comment]
Oops, one more, there's an extra closing curling brace in the Zed Java Settings codeblock

**Comment by ABckh** [comment]
Hi @notpeter, sorry for commenting after the merge, I agree with @valentinegb. For `zed-java-eclipse-jdtls`, there's no need to install jdtls manually. I think it would be helpful to mention this, as it's the main difference between the extensions.

**Comment by notpeter** [comment]
But `zed-java-eclipse-jdtls` still needs OpenJDK, right?

**Comment by notpeter** [comment]
I tried to incorporate all those notes in this follow on PR:
- https://github.com/zed-industries/zed/pull/19390

Thank you all for your work to get this documented and your continued detailed feedback.
I very much appreciate it.

**Comment by ABckh** [comment]
> But `zed-java-eclipse-jdtls` still needs OpenJDK, right?

Yes, to develop with Java, you do need a JDK (Java Development Kit) installed. OpenJDK is just one version of the JDK, so you'll need to have any JDK installed, whether it's OpenJDK or another variant.

**Pull request merged**